proto/buf/alpha/registry/v1alpha1/resolve.proto

```diff
@@ -121,6 +141,54 @@ message GetRemotePackageVersionRequest {
   LocalModuleReference module_reference = 2;
 }
 
+message GetGoVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetGoVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
+  string version = 1;
+}
+
+message GetMavenVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetMavenVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
+  string version = 1;
+}
+
+message GetNPMVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetNPMVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
+  string version = 1;
+}
+
+message GetSwiftVersionRequest {
+  // The plugin reference to resolve.
+  GetRemotePackageVersionPlugin plugin_reference = 1;
+  // The module reference to resolve.
+  LocalModuleReference module_reference = 2;
+}
+
+message GetSwiftVersionResponse {
+  // version is the resolved version to be used with the go module proxy.
+  string version = 1;
+}
+
 // GetRemotePackageVersionPlugin is a plugin reference.
 // If the version is empty, this is a reference to the latest version.
 message GetRemotePackageVersionPlugin {
```

Comment:
Does something like this not already exist as a generic type?

Reply:
https://github.com/bufbuild/buf/pull/2169#discussion_r1223481799 it does but it's got field `parameters` attached to it that I don't want to use. The decision we came to is to make this type that until [this file](https://github.com/bufbuild/buf/blob/fece8ff2cc6a184815a81551d94859384893e06c/proto/buf/alpha/registry/v1alpha1/generate.proto#L49) is deleted and the apis are removed that this will continue to exist

Reply:
Ugh it's `PluginReference` that needs to change, but that ship has sailed for this version of the package.

I'd call this `PluginVersion` and make it generic.

Reply:
[That already exists](https://github.com/bufbuild/buf/blob/6ddf4be46bfd9dfc1b3333334cf46d8c6727981a/proto/buf/alpha/registry/v1alpha1/plugin.proto#L47)

```
message PluginVersion {
  // The ID of the plugin version, which uniquely identifies the plugin version.
  // Mostly used for pagination.
  string id = 1;
  // The name of the version, i.e. "v1.4.0".
  string name = 2;
  // The name of the plugin to which this version relates.
  string plugin_name = 3;
  // The owner of the plugin to which this version relates.
  string plugin_owner = 4;
  // The full container image digest associated with this plugin version including
  // the algorithm.
  // Ref: https://github.com/opencontainers/image-spec/blob/main/descriptor.md#digests
  string container_image_digest = 5;
  // Optionally define the runtime libraries.
  repeated RuntimeLibrary runtime_libraries = 6;
}

```

Reply:
This is all stuff we need to clean up in v1, but yea not as relevant to this PR